### DISCUSSION TITLE
Porting react-const-elements to SWC

### DISCUSSION THREAD

**Discussion by JSerFeng** [swc-project/swc#7830] [I'm doing this]
After seen SWC new [constify plugin](https://github.com/swc-project/plugins/tree/main/packages/constify), I found its logic is a bit like [react-const-elements](https://babeljs.io/docs/babel-plugin-transform-react-constant-elements), so I tried to implement it using SWC, so far so good, but there are more tests to be done, it should be done soon and published as SWC WASM plugin.

If you are interesting, take a look at [here](https://github.com/web-infra-dev/swc-plugins/tree/main/crates/plugin_react_const_elements)

**Comment by kdy1** [comment]
Does someone actually use it? Just curious, it's not hard to me

**Reply by JSerFeng** [reply]
As there are many static v-nodes in usual React.js apps, personally I prefer to use it, reduces some diff time, and memory usage, better than not using it, so I ported it.

**Reply by SuperOleg39** [reply]
Also start to use this plugin, thanks a lot!

@JSerFeng what do you think, it is not possible to fix current plugin [deoptimizations](https://github.com/web-infra-dev/swc-plugins/blob/main/crates/integration_tests/tests/plugin_react_const_elements/main.rs#L64)?

**Comment by srghma** [comment]
awesome. but why I could  find this plugin in https://plugins.swc.rs/ ? not published?
